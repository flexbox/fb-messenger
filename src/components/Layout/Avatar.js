import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const AvatarWrapper = styled.img`
  margin-right: 0.5em;

  ${props =>
    props.size === 'small' &&
    css`
      clip-path: circle(10px at center);
      width: 20px;
    `}

  ${props =>
    props.size === 'medium' &&
    css`
      clip-path: circle(15px at center);
      width: 30px;
      height: 30px;
    `}

  ${props =>
    props.size === 'large' &&
    css`
      clip-path: circle(25px at center);
      width: 50px;
    `}
`;

const Avatar = ({ username, size = 'medium' }) => (
  <AvatarWrapper
    size={size}
    src={`/images/${username}_lg.jpg`}
    alt={`${username}`}
  />
);

Avatar.propTypes = {
  username: PropTypes.string,
  size: PropTypes.string
};

export default Avatar;
