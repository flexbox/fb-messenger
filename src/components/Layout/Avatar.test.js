import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Avatar from './Avatar';

// TASK 1: configure the Enzyme adapter. Hint https://airbnb.io/enzyme/

Enzyme.configure({ adapter: new Adapter() });

describe('<Avatar />', () => {
  it('renders Avatar', () => {
    const wrapper = shallow(<Avatar />);

    expect(toJson(wrapper)).toMatchSnapshot();
    // You can use console.log(wrapper.debug()) to console.log the component that you are testing
    // TASK 2: Shallow the Avatar component
    //    Hint:  https://github.com/adriantoine/enzyme-to-json#helper
    // TASK 3: Create the snapshot
  });
});
