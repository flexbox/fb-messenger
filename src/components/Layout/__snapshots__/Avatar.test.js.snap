// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> renders Avatar 1`] = `
<styled.img
  alt="undefined"
  size="medium"
  src="/images/undefined_lg.jpg"
/>
`;
